src/lib.rs

```diff
@@ -4221,6 +4255,291 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            "-Cforce-frame-pointers" => {
+                self.force_frame_pointers = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            "-Clink-dead-code" => {
+                self.link_dead_code = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                })
+            }
+            "-Cno-redzone" => {
+                self.no_redzone = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            "-Csoft-float" => {
+                self.soft_float = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            _ => {}
+        }
+    }
+
+    fn set_unstable_rustc_flag(&mut self, flag: &str) {
+        let (flag, value) = if let Some((flag, value)) = flag.split_once('=') {
+            (flag, Some(value.to_owned()))
+        } else {
+            (flag, None)
+        };
+
+        match flag {
+            "-Zbranch-protection" => self.branch_protection = value,
+            _ => {}
+        }
+    }
+
+    // Rust and clang/cc don't agree on what equivalent flags should look like either.
+    fn cc_flags(
+        &self,
+        build: &Build,
+        path: &PathBuf,
+        family: &ToolFamily,
+        target: &Target,
+    ) -> Vec<OsString> {
+        let push_if_supported = |flags: &mut Vec<OsString>, flag: OsString| {
+            if build
+                .is_flag_supported_inner(&flag, path, target)
+                .unwrap_or(false)
+            {
+                flags.push(flag);
+            } else {
+                build.cargo_output.print_warning(&format!(
+                    "Inherited flag {:?} is not supported by the currently used CC",
+                    flag
+                ));
+            }
+        };
+
+        let mut flags: Vec<OsString> = vec![];
+
+        match family {
+            ToolFamily::Clang { .. } | ToolFamily::Gnu => {
+                if let Some(value) = &self.branch_protection {
+                    push_if_supported(
+                        &mut flags,
+                        format!("-mbranch-protection={}", value.replace(",", "+")).into(),
+                    );
```

Comment:
Nice to have all of these! Now we can (in a follow-up to this PR) selectively disable it when we know that all the compilers we care about support the flag.